File: app/src/main/java/com/testproject/kaera/ringtestapp/di/components/AppComponent.java
package com.testproject.kaera.ringtestapp.di.components;

import com.testproject.kaera.ringtestapp.controllers.GalleryController;
import com.testproject.kaera.ringtestapp.controllers.TopListController;
import com.testproject.kaera.ringtestapp.di.modules.AppModule;
import com.testproject.kaera.ringtestapp.ui.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent extends PipeComponent {

    void inject(MainActivity object);

    void inject(TopListController topListController);

    void inject(GalleryController galleryController);
}

File: app/src/main/java/com/testproject/kaera/ringtestapp/di/components/AppComponent.kt
package com.testproject.kaera.ringtestapp.di.components

import com.testproject.kaera.ringtestapp.controllers.GalleryController
import com.testproject.kaera.ringtestapp.controllers.TopListController
import com.testproject.kaera.ringtestapp.di.modules.AppModule
import com.testproject.kaera.ringtestapp.ui.MainActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent : PipeComponent {

    fun inject(activity: MainActivity)

    fun inject(controller: TopListController)

    fun inject(controller: GalleryController)
}